feat: implement core REST API endpoints

Ang-m4 opened the pull request:
## Description
Corporate Partner Access API (version v1). Endpoints to manage corporate partners, their catalogs, catalog courses, catalog learners and catalog email regex patterns. Includes nested routes under catalogs that auto-inject `catalog_id` on create. Current permissions limited to `IsAuthenticated` while role scoping is refined.

## Key Changes
- Core ViewSets and routes (partners, catalogs, catalog_courses, catalog_learners, catalog_regexes) plus nested catalog actions (learners, courses, email_regexes).
- Nested POST endpoints no longer require `catalog_id` in body; it is injected from the URL path.
- Serializers unified on exposed field name `catalog_id` (`source='catalog'`).
- Position field for catalog courses (ordering groundwork; reorder endpoint TBD).

## URL Descriptions & Use Cases
API base: `/corporate_partner_access/api/v1/`

- `GET /partners/` — list partners
- `GET /partners/{partner_id}/` — retrieve partner
	- Search: `?search=<code|name>`
	- Ordering: `?ordering=name|code|id` (default: name)

- `GET|POST /partners/{partner_id}/catalogs/`
- `GET|PUT|PATCH|DELETE /partners/{partner_id}/catalogs/{catalog_id}/`
	- Filters: `?corporate_partner=<id>`, `?is_public=true|false`
	- Search: `?search=<name|slug>`
	- Ordering: `?ordering=name|id|available_start_date|available_end_date` (default: name)

- `GET|POST /partners/{partner_id}/catalogs/{catalog_id}/courses/`
- `GET|PUT|PATCH|DELETE /partners/{partner_id}/catalogs/{catalog_id}/courses/{course_id}/`
	- Filters: `?catalog=<id>`, `?course_overview=<id>`
	- Search: `?search=<course display name>`
	- Ordering: `?ordering=position|id` (default: position)

- `GET|POST /partners/{partner_id}/catalogs/{catalog_id}/learners/`
- `GET|PUT|PATCH|DELETE /partners/{partner_id}/catalogs/{catalog_id}/learners/{learner_id}/`
	- Filters: `?catalog=<id>`, `?active=true|false`, `?user=<id>`
	- Search: `?search=<username|email>`
	- Ordering: `?ordering=id|user_id` (default: id)

- `GET|POST /partners/{partner_id}/catalogs/{catalog_id}/email-regexes/`
- `GET|PUT|PATCH|DELETE /partners/{partner_id}/catalogs/{catalog_id}/email-regexes/{regex_id}/`
	- Filters: `?catalog=<id>`

## Working on
- Enhanced permission classes (partner scoping, staff-only writes, object-level checks).
- Reorder endpoint for catalog courses (bulk position update) & possible sparse positioning strategy.
- Bulk add/remove learners & courses.

Ang-m4 committed:
feat: update requirements files to include DRF dependencies

Ang-m4 committed:
feat: implement Corporate Partners API

- add ReadOnlyViewSet for Corporate Partners
- add v0 API configuration
- add serializers for Corporate Partners model

Ang-m4 committed:
feat: add CourseOverviewSerializer wrapper

Ang-m4 committed:
feat: add flex_catalog basic serializers

Ang-m4 committed:
feat: implement Corporate Partner Catalogs API

Ang-m4 committed:
feat: implement Catalog Learners and Courses API

Ang-m4 committed:
feat: enhance serializers and views for Corporate Partner API

- add error handling in get_logo_url method
- implement avaliable dates validation for CPartnerCatalog

Ang-m4 committed:
feat: implement Corporate Parnet Catalog Regex API

Ang-m4 committed:
feat: enhance Corporate Partner API with filtering and ordering

Alec4r commented:
I have been thinking about the reorder endpoint ... maybe we could use something like ```PATCH corporate-access/api/v1/catalogs:reorder``` and send a body like:

passing a complete list:
```json
{"order": [{"id": 10, "position": 1}, {"id": 7, "position": 2}, ...]}
```

or maybe just passing the new position of one element and the backend will calculate and reorder the other ones:

```json
{"moves": [{"id": 7, "position": 1}]}
```

Ang-m4 committed:
fix: url snake case formatting for APIs

Ang-m4 committed:
fix: update regex validation to avoid nested quantifiers

Ang-m4 committed:
feat: add email regexes nested endpoint to catalogs

Ang-m4 committed:
refactor: update API v0 to v1

Ang-m4 committed:
feat: implement nested routing for Corporate Partner API

- add minimal User serializer
- add InjectCatalogMixin to override the catalog id at nested endpoints

Ang-m4 committed:
feat: implement partners nested routing

- create a nested mixin to override the data with fields pk

Alec4r committed:
feat: add swagger docs
